tiled/server/dependencies.py

```diff
@@ -29,113 +22,112 @@ def get_root_tree():
     )
 
 
-def get_entry(structure_families: Optional[set[StructureFamily]] = None):
-    async def inner(
-        path: str,
-        request: Request,
-        security_scopes: SecurityScopes,
-        principal: Union[Principal, SpecialUsers] = Depends(get_current_principal),
-        authn_scopes: Scopes = Depends(get_current_scopes),
-        root_tree: pydantic_settings.BaseSettings = Depends(get_root_tree),
-        session_state: dict = Depends(get_session_state),
-        _=Security(check_scopes),
-    ) -> MapAdapter:
-        """
-        Obtain a node in the tree from its path.
-
-        Walk down the path starting from the root of the tree and filter
-        access by the specified scopes.
-
-        session_state is an optional dictionary passed in the session token
-        """
-        path_parts = [segment for segment in path.split("/") if segment]
-        entry = root_tree
-        access_policy = getattr(request.app.state, "access_policy", None)
-
-        # If the entry/adapter can take a session state, pass it in.
-        # The entry/adapter may return itself or a different object.
-        if hasattr(entry, "with_session_state") and session_state:
-            entry = entry.with_session_state(session_state)
-        # start at the root
-        # filter and keep only what we are allowed to see from here
-        entry = await filter_for_access(
-            entry,
-            access_policy,
-            principal,
-            authn_scopes,
-            ["read:metadata"],
-            request.state.metrics,
-        )
-        try:
-            for i, segment in enumerate(path_parts):
-                if hasattr(entry, "lookup_adapter"):
-                    # New catalog adapter
-                    # This adapter can jump directly to the node of interest,
-                    # but currenty doesn't, to ensure access_policy is applied.
-                    # Raises NoEntry or BrokenLink if the path is not found
-                    entry = await entry.lookup_adapter([segment])
-                else:
-                    # Old-style dict-like interface
-                    # Traverse into sub-tree(s) to reach the desired entry
-                    try:
-                        entry = entry[segment]
-                    except (KeyError, TypeError):
-                        raise NoEntry(path_parts)
+async def get_entry(
+    path: str,
+    security_scopes: List[str],
+    principal: Union[Principal, SpecialUsers],
+    authn_scopes: Scopes,
+    root_tree: pydantic_settings.BaseSettings,
+    session_state: dict,
+    structure_families: Optional[set[StructureFamily]] = None,
+    metrics: Optional[dict] = None,
```

Comment:
It seems like we could make this required, now, and thus simplify the code below. I believe at the time this was drafted, we were unsure.

Reply:
Fixed in: https://github.com/bluesky/tiled/pull/995/commits/0dca9aa0f8754e9587de69259d07e76ca592cec3